VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participants-panel/participants-panel.component.html

```diff
@@ -119,7 +119,12 @@ <h2 class="govuk-heading-s white">{{ 'participants-panel.participants' | transla
               *ngIf="!participant.transferringIn"
               appTooltip
               [text]="
-                (participant.isAvailable() ? 'participants-panel.call-witness' : 'participants-panel.witness-unavailable') | translate
+                (participant.isAvailable() ? 
+                  (isWitness(participant) ? 
+                    ('participants-panel.call-witness' | translate) : 
+                    ('participants-panel.admit-participant' | translate : {role: 'hearing-role.' + (participant.hearingRole | hyphenate) | translate | lowercase})
+                  ) : 
+                  'participants-panel.participant-unavailable' | translate : {role: 'hearing-role.' + (participant.hearingRole | hyphenate) | translate})
```

Comment:
participant.isWitness would not work for me and always returned the admit-participant value. Not sure why. Any insight/advice/thoughts on this would be appreciated as I don't like having this additional method.

Reply:
Although this has been done a similar way in other components.

Reply:
I am not sure why that wouldn't work because if participant was null it would just show unavailable text instead. So does the unit test fail when you switch from using this function to particiapnt.isWitness? 

Also I was wondering if we could refactor this block completely into the typescript file, move all the logic inside of a function that returns the expected text instead as having if condition this way is a bit difficult to read. I am not sure if we get any benefits in having this all in the template file.

Reply:
It's preferred to use pipes for a few reasons. Performance is one. If the pipes are pure they will only be called when the input changes as opposed to every change detection cycle. I don't know for sure if TranslatePipe is pure but the other 2 are.
 
The other is if we have it in the typescript file we need to worry about subscribing and unsubscribing from different observables. By using pipes that is all taken care of for us.

Reply:
I will say the layout here isn't great. Since it's a directive input I'm not sure if there's a nicer way to do it. I don't think a function will work as the pipes vary for each case, but I'll have a little think.

Reply:
I'm afraid I can't come up with a nicer way to do this as the only way I can think would be to use *ngIf with ng-templates but that would involve duplicating the other attributes which I think would not be worth it.